Remove the `rustc-serialize` crate and use the `hex` crate instead.

jeandudey opened the pull request.

svyatonik commented:
There are some compilation errors when tests are running (`cargo test --all`). Could you please fix?

jeandudey commented:
@svyatonik done :+1:

jeandudey commented:
@svyatonik Thanks for the review, I addressed some of your changes and removed `hex` in favor of `rustc-hex`, however using `hex-literal` for tests seems to be an overkill both for maintenance and compile times, and also adds some confusion (two crates to parse hexadecimal strings?).

svyatonik commented on chain/src/transaction.rs:
```diff
@@ -100,7 +100,8 @@ pub struct Transaction {
 impl From<&'static str> for Transaction {
 	fn from(s: &'static str) -> Self {
-		deserialize(&s.from_hex().unwrap() as &[u8]).unwrap()
+        let vec: Vec<u8> = s.from_hex().unwrap();
```
Please use tabs

svyatonik commented on primitives/src/bytes.rs:
```diff
@@ -84,7 +84,8 @@ impl io::Write for Bytes {
 impl fmt::Debug for Bytes {
 	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-		f.write_str(&self.0.to_hex())
+        let s: String = self.0.to_hex();
```
Please use tabs

svyatonik commented on primitives/src/hash.rs:
```diff
@@ -81,13 +81,15 @@ macro_rules! impl_hash {
 		impl fmt::Debug for $name {
 			fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-				f.write_str(&self.0.to_hex())
+                let s: String = self.0.to_hex();
```
Please use tabs

svyatonik commented on primitives/src/hash.rs:
```diff
@@ -81,13 +81,15 @@ macro_rules! impl_hash {
 		impl fmt::Display for $name {
 			fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-				f.write_str(&self.0.to_hex())
+                let s: String = self.0.to_hex();
```
Please use tabs

svyatonik commented on test-data/src/block.rs:
```diff
@@ -312,6 +312,7 @@ impl<F> TransactionBuilder<F> where F: Invoke<chain::Transaction> {
 			//0xfd...(0xffff + 3) => 3,
 			0xfd...0x10002 => 3,
 			//0x10000...(0xffff_ffff + 5) => 5,
+            #[cfg(target_pointer_width = "64")]
```
Is that intentional?

jeandudey replied:
Ooops no, let me delete it, it was a one-time hack to build on i686. That constant overflows `usize`, that's content for another PR.
